build/TestNuGetsTasks/TestMonoGameTemplateTaskBase.cs

```diff
@@ -0,0 +1,597 @@
+        {
+            context.StartProcess("dotnet", new ProcessSettings
+            {
+                Arguments = "new uninstall MonoGame.Templates.CSharp"
+            });
+        }
+        catch
+        {
+            // Template might not be installed, continue
+            context.Information("No existing MonoGame templates found to uninstall.");
+        }
+    }
+
+    private void InstallTemplates(BuildContext context, string templateVersion, string nugetSourcePath)
+    {
+        context.Information($"Installing MonoGame templates version {templateVersion}...");
+        context.StartProcess("dotnet", new ProcessSettings
+        {
+            Arguments = $"new install MonoGame.Templates.CSharp::{templateVersion} --nuget-source \"{nugetSourcePath}\""
+        });
+    }
+
+    private void CreateTestProject(BuildContext context, string projectPath)
+    {
+        context.Information($"Creating new MonoGame {TemplateName} project in: {projectPath}");
+        context.CreateDirectory(projectPath);
+        context.StartProcess("dotnet", new ProcessSettings
+        {
+            Arguments = $"new {TemplateShortName} -n TestProject",
+            WorkingDirectory = projectPath
+        });
+    }
+
+    private void ReplaceDotnetToolsConfig(BuildContext context, string projectDir, string version)
+    {
+        var configDir = IOPath.Combine(projectDir, ".config");
+        var dotnetToolsPath = IOPath.Combine(configDir, "dotnet-tools.json");
+        
+        context.Information($"Replacing dotnet-tools.json with platform-specific version for: {context.Environment.Platform.Family}");
+        
+        if (!Directory.Exists(configDir))
+        {
+            context.CreateDirectory(configDir);
+        }
+
+        // Always replace with platform-specific version to avoid cross-platform tool issues
+        var toolsJson = GetPlatformSpecificToolsJson(context, version);
+        File.WriteAllText(dotnetToolsPath, toolsJson);
+        
+        context.Information("Platform-specific dotnet-tools.json created successfully.");
+    }
+
+    private string GetPlatformSpecificToolsJson(BuildContext context, string version)
+    {
+        var platform = context.Environment.Platform.Family;
+        
+        // Base tools that are available on all platforms
+        var baseTools = $$"""
+        {
+          "version": 1,
+          "isRoot": true,
+          "tools": {
+            "dotnet-mgcb": {
+              "version": "{{version}}",
+              "commands": [
+                "mgcb"
+              ]
+            },
+            "dotnet-mgcb-editor": {
+              "version": "{{version}}",
+              "commands": [
+                "mgcb-editor"
+              ]
+            }
+        """;
+
+        // Add platform-specific editor tool
+        string platformSpecificTool = platform switch
+        {
+            PlatformFamily.Windows => $$"""
+            ,
+            "dotnet-mgcb-editor-windows": {
```

Comment:
Ident this and lines above one tab to the right.

Reply:
Indented